### PR TITLE
Add Function CR documentation

### PR DESCRIPTION
**Pull request by bszwarc**
**Description**

Changes proposed in this pull request:

- Add Function CR documentation
- Add Knative section to Serverless
- Improve Serverless Architecture doc and Overview
- Improve lambda functions documentation

**Related issue(s)**
See also #5129

### CODE REVIEW

File: docs/serverless/06-01-function.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+The `functions.serverless.kyma-project.io` CustomResourceDefinition (CRD) is a detailed description of the kind of data and the format used to manage lambda functions within Kyma. To get the up-to-date CRD and show the output in the `yaml` format, run this command:
+
+```
+kubectl get crd functions.serverless.kyma-project.io -o yaml
+```
+
+## Sample custom resource
+
+The following Function object creates a lambda function which runs on Node.js 8 runtime and responds to HTTP requests with "Hello World". It has low requirements in terms of compute resources and is therefore classified as size S.
```
Comment (klaudiagrz) [inline docs/serverless/06-01-function.md]:
```suggestion
The following Function object creates a lambda function which runs on the Node.js 8 runtime and responds to HTTP requests with "Hello World". It has low requirements in terms of computing resources and is therefore classified as size S.
```

File: docs/serverless/06-01-function.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+```
+
+## Custom resource properties
+
+This table lists all the possible properties of a given resource together with their descriptions:
+
+| Property | Mandatory | Description |
+|----------|:---------:|-------------|
+| **metadata.name** | YES | Specifies the nme of the CR. |
```
Comment (klaudiagrz) [inline docs/serverless/06-01-function.md]:
```suggestion
| **metadata.name** | YES | Specifies the name of the CR. |
```

File: docs/serverless/06-01-function.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+| **spec.function** | YES | Provides the source code of the lambda function. |
+| **spec.functionContentType** | YES | Specifies the content type of the function's code defined in the **function** property. The content type can be plaintext or base64-encoded. |
+| **spec.runtime** | YES | Specifies the software runtime used to run the function's code. |
+| **spec.size** | YES | Specifies the compute requirement of the function expressed in size, such as S, M, L or XL. |
```
Comment (klaudiagrz) [inline docs/serverless/06-01-function.md]:
```suggestion
| **spec.size** | YES | Specifies the computing requirement of the function expressed in size, such as S, M, L, or XL. |
```

File: docs/serverless/06-01-function.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+| **spec.deps** | NO | Specifies the dependencies of the lambda function. |
+| **spec.env** | NO | Specifies environment variables you need to export for the lambda function. |
+| **spec.timeout** | NO | Specifies the duration in seconds after which the function execution is terminated.The default value is 180. |
```
Comment (klaudiagrz) [inline docs/serverless/06-01-function.md]:
```suggestion
| **spec.timeout** | NO | Specifies the duration in seconds after which the function execution is terminated. The default value is `180`. |
```

File: docs/serverless/06-01-function.md (hunk 1 above)
Comment (bszwarc) [reply docs/serverless/06-01-function.md]: Done

File: docs/serverless/06-01-function.md (hunk 3 above)
Comment (bszwarc) [reply docs/serverless/06-01-function.md]: Done

File: docs/serverless/06-01-function.md (hunk 2 above)
Comment (bszwarc) [reply docs/serverless/06-01-function.md]: Done.

**Pull request merged**